File: others/haskell/const.hs
main = do
  let foo = "string"
      bar = 3.4
      baz = True
  -- 標準入力を読み込む
  c <- getContents
  putStrLn foo
  putStrLn (show bar) -- `show`は文字列以外のものを文字列に変換する関数
  print baz -- `print`はshowをしてから`putStrLn`する関数
  putStrLn c

File: others/ruby/const.rb
#!/usr/bin/env ruby

# 定数参照の優先順位
class Foo
  Const = 'foo'
end

class Bar
  Const = 'Bar'
  class Baz < Foo
    p Const  # -> "Bar" 外側の定数
    # 親クラスの定数は明示的に指定しなければ見えない
    p Foo::Const # -> "Foo"
  end
end
